src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsHelper.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal static class StaticWebAssetsHelper
+    {
+        private const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+        private const string ManifestRootElementName = "StaticWebAssets";
+        private const string VersionAttributeName = "Version";
+        private const string ContentRootElementName = "ContentRoot";
+
+        internal static void UseStaticWebAssets(this IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(this IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var staticWebAssetContentRoots = Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (staticWebAssetContentRoots.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                staticWebAssetContentRoots.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(staticWebAssetContentRoots);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            var assembly = Assembly.Load(environment.ApplicationName);
+            if (assembly.GetManifestResourceNames().Any(a => a == StaticWebAssetsManifestName))
+            {
+                return assembly.GetManifestResourceStream(StaticWebAssetsManifestName);
+            }
+            else
+            {
+                // Fallback to physical file as we plan to use a file on disk instead of the embedded resource.
+                var filePath = Path.Combine(Path.GetDirectoryName(new Uri(assembly.CodeBase).LocalPath), $"{environment.ApplicationName}.StaticWebAssets.xml");
```

Comment:
Can we resolve what we're doing here before checking this in? I'm really not sold on the consequences of making this an embedded resource. 

I would also expect to see `.Location` here instead of codebase and URI if this is needed can you elaborate?

Reply:
@rynowak The embedded resource part is going away, but I won't have time for all the dependencies to flow from the Razor SDK before I check this in, so this is just a tactical implementation until I can take the embedded resource part in preview7 and simply rely on the physical file.

https://blogs.msdn.microsoft.com/suzcook/2003/06/26/assembly-codebase-vs-assembly-location/

https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/ApplicationParts/RelatedAssemblyAttribute.cs#L116

I will change the implementation to do what we do in Razor

Reply:
Updated